source/modules/MEI.stringtab.xml

```diff
@@ -1,51 +1,91 @@
+    <classes>
+      <memberOf key="att.stringtab.vis" />
+    </classes>
+  </classSpec>
   <classSpec ident="att.stringtab" module="MEI.stringtab" type="atts">
     <desc xml:lang="en">String tablature string and fret information.</desc>
     <attList>
+      <attDef ident="tab.align" usage="opt">
+        <desc xml:lang="en">Attribute that describes the vertical alignment of tablature symbols.
+          Only applicable in cases where the symbols' vertical position does not communicate other
+          information, such as courses (i.e., only in German lute tablature). Typical values are <val>
+            top</val> and <val>bottom</val>.</desc>
+        <datatype>
+          <rng:ref name="data.VERTICALALIGNMENT" />
+        </datatype>
+      </attDef>
+      <!-- Below here is unchecked / legacy -->
       <attDef ident="tab.fing" usage="opt">
-        <desc xml:lang="en">Indicates which finger, if any, should be used to play an individual string. The
-          index, middle, ring, and little fingers are represented by the values 1-4, while <val>t</val> is
+        <desc xml:lang="en">This attribute is deprecated and will be removed in a future version.
+          Indicates which finger, if any, should be used to play an individual string. The index,
+          middle, ring, and little fingers are represented by the values 1-4, while <val>t</val> is
           for the thumb. The values <val>x</val> and <val>o</val> indicate muffled and open strings,
           respectively.</desc>
         <datatype>
-          <rng:ref name="data.FINGER.FRET"/>
+          <rng:ref name="data.FINGER.FRET" />
         </datatype>
       </attDef>
       <attDef ident="tab.fret" usage="opt">
         <desc xml:lang="en">Records the location at which a string should be stopped against a fret.</desc>
         <datatype>
-          <rng:ref name="data.FRETNUMBER"/>
+          <rng:ref name="data.FRETNUMBER" />
+        </datatype>
+      </attDef>
+      <attDef ident="tab.line" usage="opt">
+        <desc xml:lang="en">Used in German lute tablature in cases where vertical positioning deviates from the norm which can be specified by <att>tab.align</att>. 
+        Indicates the position of the note on one of the horizontal strands corresponding to the <att>lines</att> attribute on <gi>staffDef</gi>. 
+        (Note that in this case, the lines are conceptual rather than visible).
+        </desc>
+        <datatype>
+          <rng:ref name="data.CLEFLINE" />
         </datatype>
       </attDef>
       <attDef ident="tab.string" usage="opt">
-        <desc xml:lang="en">Records which string is to be played.</desc>
+        <desc xml:lang="en">This attribute is deprecated in favour of <att>tab.course</att> and will
+          be removed in a future version. Records which string is to be played.</desc>
+        <datatype>
+          <rng:ref name="data.STRINGNUMBER" />
```

Comment:
There is literally no difference between `data.STRINGNUMBER` and `data.COURSENUMBER` except for their name. Unify to `data.STRINGORCOURSENUMBER` (or similar)?

Reply:
I think the idea is to remove `data.STRINGNUMBER` in MEI 6.0.

Reply:
I see. Maybe a short deprecation notice in `data.STRINGNUMBER` could be helpful, too?

Reply:
I tried to use `@validUntil` but was not successful. Any ideas how this should be used? Alternatively we can add an XML comment, or something in the `desc`. What do you think?

Reply:
Yes, I think it would be sufficient like in the desc in the snippet here above: "This attribute is deprecated in favour ... and will be removed in a future version. ..."

Reply:
Deprectation notice added.